src/main/java/techreborn/tiles/generator/TileSolarPanel.java

```diff
@@ -44,61 +45,64 @@
 @RebornRegistry(modID = ModInfo.MOD_ID)
 public class TileSolarPanel extends TilePowerAcceptor implements IToolDrop {
 
-	@ConfigRegistry(config = "machines", category = "solar_panel", key = "SolarPanelMaxOutput", comment = "Solar Panel Max Output (Value in EU)")
-	public static int maxOutput = 32;
-	@ConfigRegistry(config = "machines", category = "solar_panel", key = "SolarPanelMaxEnergy", comment = "Solar Panel Max Energy (Value in EU)")
-	public static int maxEnergy = 1000;
-	@ConfigRegistry(config = "machines", category = "solar_panel", key = "SolarPanelEnergyPerTick", comment = "Solar Panel Energy Per Tick (Value in EU)")
-	public static int energyPerTick = 1;
+	//	@ConfigRegistry(config = "machines", category = "solar_panel", key = "SolarPanelMaxOutput", comment = "Solar Panel Max Output (Value in EU)")
+	//	public static int maxOutput = 32;
+	//	@ConfigRegistry(config = "machines", category = "solar_panel", key = "SolarPanelMaxEnergy", comment = "Solar Panel Max Energy (Value in EU)")
+	//	public static int maxEnergy = 1000;
+	//	@ConfigRegistry(config = "machines", category = "solar_panel", key = "SolarPanelEnergyPerTick", comment = "Solar Panel Energy Per Tick (Value in EU)")
+	//	public static int energyPerTick = 1;
 
 	boolean shouldMakePower = false;
 	boolean lastTickSate = false;
-
 	int powerToAdd;
+	private int generationRateD = 0;
+	private int generationRateN = 0;
+	private int internalCapacity = 0;
+	private int generationCurrent = 0;
+	private EnumPowerTier tier = null;
 
 	public TileSolarPanel() {
 		super();
 	}
 
 	@Override
+	public void onLoad() {
+		EnumPanelType panel = getPanelType();
+		this.generationRateD = panel.generationRateD;
+		this.generationRateN = panel.generationRateN;
+		this.internalCapacity = panel.internalCapacity;
+		this.tier = panel.powerTier;
+	}
+
+	@Override
 	public void update() {
 		super.update();
-		if (!this.world.isRemote) {
-			if (this.world.getTotalWorldTime() % 60 == 0) {
-				this.shouldMakePower = this.isSunOut();
+		if (this.world.isRemote) { return; }
 
-			}
-			if (this.shouldMakePower) {
-				this.powerToAdd = energyPerTick;
+			if (isSunOut()) {
+				this.powerToAdd = this.generationRateD;
+				this.addEnergy(this.powerToAdd);
+			} else if(this.world.canBlockSeeSky(this.pos.up())) {
+				this.powerToAdd = this.generationRateN;
 				this.addEnergy(this.powerToAdd);
```

Comment:
You can addEnergy after if block

Reply:
If you do it after It doesn't account for it being blocked. So if a block was above it, it wouldn't stop generating power.